domain/schema/model/sql/0015-charm.sql

```diff
@@ -101,11 +104,25 @@ CREATE TABLE charm_origin (
     REFERENCES charm (uuid)
 );
 
+CREATE UNIQUE INDEX idx_charm_origin_reference_name_revision
+ON charm_origin (reference_name, revision);
+
+CREATE VIEW v_charm_origin AS
+SELECT
+    co.charm_uuid,
+    co.reference_name,
+    cs.name AS source,
+    co.charmhub_identifier,
+    co.revision,
+    co.version
+FROM charm_origin AS co
+LEFT JOIN charm_source AS cs ON co.source_id = cs.id;
```

Comment:
If `co.source_id` is `NOT NULL` and has a FK to `charm_source`, this is just a `JOIN`; you are never without the row.